commands/project/repo.go

```diff
@@ -9,16 +11,40 @@ import (
 	repoCmdDelete "github.com/profclems/glab/commands/project/delete"
 	repoCmdFork "github.com/profclems/glab/commands/project/fork"
 	repoCmdSearch "github.com/profclems/glab/commands/project/search"
+	"github.com/profclems/glab/internal/utils"
 
 	"github.com/spf13/cobra"
 )
 
+type RepoOpts struct {
+	OpenInBrowser bool
+}
+
 func NewCmdRepo(f *cmdutils.Factory) *cobra.Command {
+	opts := RepoOpts{}
 	var repoCmd = &cobra.Command{
 		Use:     "repo <command> [flags]",
 		Short:   `Work with GitLab repositories and projects`,
 		Long:    ``,
 		Aliases: []string{"project"},
+		RunE: func(cmd *cobra.Command, args []string) error {
+			cfg, err := f.Config()
+			if err != nil {
+				return err
+			}
+			baseRepo, err := f.BaseRepo()
+			if err != nil {
+				return err
+			}
+			if opts.OpenInBrowser {
+				browser, err := cfg.Get(baseRepo.RepoHost(), "browser")
+				if err != nil {
+					return err
+				}
+				return utils.OpenInBrowser(fmt.Sprintf("https://%s/%s", baseRepo.RepoHost(), baseRepo.FullName()), browser)
+			}
+			return nil
+		},
```

Comment:
Instead of using the `RunE`, you might want to consider using a `PreRun` hook instead.
Using the `RunE` will prevent it from executing child commands.

Reply:
```suggestion
		PreRunE: func(cmd *cobra.Command, args []string) error {
			if opts.OpenInBrowser {
				cfg, err := f.Config()
				if err != nil {
					return err
				}
				baseRepo, err := f.BaseRepo()
				if err != nil {
					return err
				}
				browser, err := cfg.Get(baseRepo.RepoHost(), "browser")
				if err != nil {
					return err
				}
				return utils.OpenInBrowser(fmt.Sprintf("https://%s/%s", baseRepo.RepoHost(), baseRepo.FullName()), browser)
			}
			return nil
		},
```

Reply:
For the subcommands, you can do that directly in the `RunE`